mat64/dense_arithmetic.go

```diff
@@ -361,9 +362,12 @@ func (m *Dense) Mul(a, b Matrix) {
 		panic(ErrShape)
 	}
 
-	var w Dense
+	var w *Dense
 	if m != a && m != b {
-		w = *m
+		w = m
+	} else {
+		m.reuseAs(ar, bc)
```

Comment:
You are right, I did misunderstand. Ignore the comment about the inner test. Sorry about that.